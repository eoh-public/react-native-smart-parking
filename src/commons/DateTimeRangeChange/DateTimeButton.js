import React, { memo } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { IconFill } from '@ant-design/icons-react-native';
import moment from 'moment';

import Text from '../Text';
import { Colors } from '../../configs';

const DateTimeButton = memo(({ onPress, time, date, formatType }) => {
  let format = 'DD.MM.YY hh:mm';
  if (formatType === 'date') {
    format = 'DD.MM.YY';
  }
  return (
    <TouchableOpacity style={styles.dateSelect} onPress={onPress}>
      <Text type={'Label'} color={Colors.Gray8} style={styles.txtTime}>
        {moment(time).format(format)}
      </Text>
      <IconFill name={'caret-down'} size={16} color={Colors.Gray8} />
    </TouchableOpacity>
  );
});

export default DateTimeButton;

const styles = StyleSheet.create({
  dateSelect: {
    paddingLeft: 12,
    paddingVertical: 8,
    paddingRight: 8,
    borderRadius: 2,
    borderWidth: 1,
    borderColor: Colors.Gray5,
    marginHorizontal: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: 124,
    alignItems: 'center',
  },
  txtTime: {
    marginRight: 4,
  },
});
