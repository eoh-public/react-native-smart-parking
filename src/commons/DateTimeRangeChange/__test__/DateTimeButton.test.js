import React from 'react';
import renderer, { act } from 'react-test-renderer';
import moment from 'moment';
import DateTimeButton from '../DateTimeButton';

describe('Test DateTimeButton', () => {
  const time = moment('2021-01-20T05:00:00.629Z').utcOffset(0);
  let tree;

  test('render DateTimeButton formatType = "date"', () => {
    act(() => {
      tree = renderer.create(
        <DateTimeButton onPress={() => {}} time={time} formatType="date" />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render DateTimeButton formatType is undefined', () => {
    act(() => {
      tree = renderer.create(<DateTimeButton onPress={() => {}} time={time} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
