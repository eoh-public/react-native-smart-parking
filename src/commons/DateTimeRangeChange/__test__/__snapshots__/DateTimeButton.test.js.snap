// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateTimeButton render DateTimeButton formatType = "date" 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#D9D9D9",
      "borderRadius": 2,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 8,
      "opacity": 1,
      "paddingLeft": 12,
      "paddingRight": 8,
      "paddingVertical": 8,
      "width": 124,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginRight": 4,
        },
      ]
    }
  >
    20.01.21
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "antfill",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Test DateTimeButton render DateTimeButton formatType is undefined 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#D9D9D9",
      "borderRadius": 2,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 8,
      "opacity": 1,
      "paddingLeft": 12,
      "paddingRight": 8,
      "paddingVertical": 8,
      "width": 124,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginRight": 4,
        },
      ]
    }
  >
    20.01.21 05:00
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "antfill",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    
  </Text>
</View>
`;
